email_totals/app.py

```diff
@@ -84,17 +159,97 @@ def get_account_totals(target_p, compare_p, minimum):
             222233334444:
                 total: 10
     ```
+
+    ```
+    111122223333: friendly-name
+    222233334444: account-two
+    ```
     """
 
+    def _build_result_dict(results_by_time):
+        """
+        Transform the results from cost explorer into a dictionary of the form:
+        ```
+        111122223333: 100.0
+        222233334444: 10
+        ```
+        """
+        account_totals = {}
+        for result in results_by_time:
+            for group in result['Groups']:
+                amount = float(group['Metrics']['UnblendedCost']['Amount'])
+
+                # Keys preserve the order defined in the GroupBy parameter from
+                # the call to get_cost_and_usage().
+                if len(group['Keys']) != 1:
+                    LOG.error(f"Unexpected grouping: {group['Keys']}")
+                    continue
+
+                # Add this account total to our output
+                account_id = group['Keys'][0]
+                if account_id not in account_totals:
+                    account_totals[account_id] = amount
+                else:
+                    LOG.error(f"Duplicate account total found: {account_id}")
+
+        return account_totals
+
+    def _build_attr_dict(attributes):
+        """
+        Transform DimensionValueAttributes to a useful dict
```

Comment:
i don't understand what this is doing.  what makes the transformed dict more useful?  would you be able to provide a little more docs here?

Reply:
oh, that was a poorly worded docstring, I've updated it